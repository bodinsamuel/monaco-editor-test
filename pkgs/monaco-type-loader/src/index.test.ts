import path from 'path';
import { load } from '.';

it('should load', async () => {
  const res = await load({
    rootDir: path.join(__dirname, '..', '..', '..'),
    entries: [
      'cheerio',
      '@types/prettier',
      path.join(__dirname, '/testData/testData.d.ts'),
    ],
    pathNodeModules: path.join(__dirname, '../../../node_modules/'),
    pathToWrite: 'gen.ts',
    logger: null,
  });
  expect(res).toMatchSnapshot();
});
