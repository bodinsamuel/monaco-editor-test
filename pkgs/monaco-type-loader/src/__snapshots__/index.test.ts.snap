// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should load and generate file 1`] = `
"/* eslint-disable @typescript-eslint/no-var-requires, import/order, global-require */
// This is a generated file, do not edit.
// Built with @h1fra/monaco-type-loader; ts version v0.0.0

import * as MonacoEditor from 'monaco-editor/esm/vs/editor/editor.api';

interface Entry {
  path: string;
  source: any;
  pkg: string | false;
}

// prettier-ignore
export const entries: Entry[] = [
  { pkg: \\"cheerio\\", path: \\"/node_modules/cheerio/package.json\\", source: require(\\"!!raw-loader!/node_modules/cheerio/package.json\\") },
{ pkg: \\"@types/prettier\\", path: \\"/node_modules/@types/prettier/package.json\\", source: require(\\"!!raw-loader!/node_modules/@types/prettier/package.json\\") },
{ pkg: false, path: \\"/pkgs/monaco-type-loader/src/testData/testData.d.ts\\", source: require(\\"!!raw-loader!/pkgs/monaco-type-loader/src/testData/testData.d.ts\\").default },
{ pkg: \\"cheerio\\", path: \\"/node_modules/cheerio/lib/index.d.ts\\", source: require(\\"!!raw-loader!/node_modules/cheerio/lib/index.d.ts\\").default },
{ pkg: \\"@types/prettier\\", path: \\"/node_modules/@types/prettier/index.d.ts\\", source: require(\\"!!raw-loader!/node_modules/@types/prettier/index.d.ts\\").default },
{ pkg: \\"cheerio\\", path: \\"/node_modules/cheerio/lib/cheerio.d.ts\\", source: require(\\"!!raw-loader!/node_modules/cheerio/lib/cheerio.d.ts\\").default },
{ pkg: \\"cheerio\\", path: \\"/node_modules/cheerio/lib/types.d.ts\\", source: require(\\"!!raw-loader!/node_modules/cheerio/lib/types.d.ts\\").default },
{ pkg: \\"cheerio\\", path: \\"/node_modules/cheerio/lib/options.d.ts\\", source: require(\\"!!raw-loader!/node_modules/cheerio/lib/options.d.ts\\").default },
{ pkg: \\"domhandler\\", path: \\"/node_modules/domhandler/package.json\\", source: require(\\"!!raw-loader!/node_modules/domhandler/package.json\\") },
{ pkg: \\"cheerio\\", path: \\"/node_modules/cheerio/lib/load.d.ts\\", source: require(\\"!!raw-loader!/node_modules/cheerio/lib/load.d.ts\\").default },
{ pkg: \\"cheerio\\", path: \\"/node_modules/cheerio/lib/static.d.ts\\", source: require(\\"!!raw-loader!/node_modules/cheerio/lib/static.d.ts\\").default },
{ pkg: \\"cheerio\\", path: \\"/node_modules/cheerio/lib/api/attributes.d.ts\\", source: require(\\"!!raw-loader!/node_modules/cheerio/lib/api/attributes.d.ts\\").default },
{ pkg: \\"cheerio\\", path: \\"/node_modules/cheerio/lib/api/traversing.d.ts\\", source: require(\\"!!raw-loader!/node_modules/cheerio/lib/api/traversing.d.ts\\").default },
{ pkg: \\"cheerio\\", path: \\"/node_modules/cheerio/lib/api/manipulation.d.ts\\", source: require(\\"!!raw-loader!/node_modules/cheerio/lib/api/manipulation.d.ts\\").default },
{ pkg: \\"cheerio\\", path: \\"/node_modules/cheerio/lib/api/css.d.ts\\", source: require(\\"!!raw-loader!/node_modules/cheerio/lib/api/css.d.ts\\").default },
{ pkg: \\"cheerio\\", path: \\"/node_modules/cheerio/lib/api/forms.d.ts\\", source: require(\\"!!raw-loader!/node_modules/cheerio/lib/api/forms.d.ts\\").default },
{ pkg: \\"htmlparser2\\", path: \\"/node_modules/htmlparser2/package.json\\", source: require(\\"!!raw-loader!/node_modules/htmlparser2/package.json\\") },
{ pkg: \\"domhandler\\", path: \\"/node_modules/domhandler/lib/index.d.ts\\", source: require(\\"!!raw-loader!/node_modules/domhandler/lib/index.d.ts\\").default },
{ pkg: \\"htmlparser2\\", path: \\"/node_modules/htmlparser2/lib/index.d.ts\\", source: require(\\"!!raw-loader!/node_modules/htmlparser2/lib/index.d.ts\\").default },
{ pkg: \\"domhandler\\", path: \\"/node_modules/domhandler/lib/node.d.ts\\", source: require(\\"!!raw-loader!/node_modules/domhandler/lib/node.d.ts\\").default },
{ pkg: \\"htmlparser2\\", path: \\"/node_modules/htmlparser2/lib/Parser.d.ts\\", source: require(\\"!!raw-loader!/node_modules/htmlparser2/lib/Parser.d.ts\\").default },
{ pkg: \\"htmlparser2\\", path: \\"/node_modules/htmlparser2/lib/Tokenizer.d.ts\\", source: require(\\"!!raw-loader!/node_modules/htmlparser2/lib/Tokenizer.d.ts\\").default },
{ pkg: \\"domelementtype\\", path: \\"/node_modules/domelementtype/package.json\\", source: require(\\"!!raw-loader!/node_modules/domelementtype/package.json\\") },
{ pkg: \\"htmlparser2\\", path: \\"/node_modules/htmlparser2/lib/FeedHandler.d.ts\\", source: require(\\"!!raw-loader!/node_modules/htmlparser2/lib/FeedHandler.d.ts\\").default },
{ pkg: \\"domutils\\", path: \\"/node_modules/domutils/package.json\\", source: require(\\"!!raw-loader!/node_modules/domutils/package.json\\") },
{ pkg: \\"domelementtype\\", path: \\"/node_modules/domelementtype/lib/index.d.ts\\", source: require(\\"!!raw-loader!/node_modules/domelementtype/lib/index.d.ts\\").default },
{ pkg: \\"domutils\\", path: \\"/node_modules/domutils/lib/index.d.ts\\", source: require(\\"!!raw-loader!/node_modules/domutils/lib/index.d.ts\\").default },
{ pkg: \\"domutils\\", path: \\"/node_modules/domutils/lib/stringify.d.ts\\", source: require(\\"!!raw-loader!/node_modules/domutils/lib/stringify.d.ts\\").default },
{ pkg: \\"domutils\\", path: \\"/node_modules/domutils/lib/traversal.d.ts\\", source: require(\\"!!raw-loader!/node_modules/domutils/lib/traversal.d.ts\\").default },
{ pkg: \\"domutils\\", path: \\"/node_modules/domutils/lib/manipulation.d.ts\\", source: require(\\"!!raw-loader!/node_modules/domutils/lib/manipulation.d.ts\\").default },
{ pkg: \\"domutils\\", path: \\"/node_modules/domutils/lib/querying.d.ts\\", source: require(\\"!!raw-loader!/node_modules/domutils/lib/querying.d.ts\\").default },
{ pkg: \\"domutils\\", path: \\"/node_modules/domutils/lib/legacy.d.ts\\", source: require(\\"!!raw-loader!/node_modules/domutils/lib/legacy.d.ts\\").default },
{ pkg: \\"domutils\\", path: \\"/node_modules/domutils/lib/helpers.d.ts\\", source: require(\\"!!raw-loader!/node_modules/domutils/lib/helpers.d.ts\\").default },
{ pkg: \\"dom-serializer\\", path: \\"/node_modules/dom-serializer/package.json\\", source: require(\\"!!raw-loader!/node_modules/dom-serializer/package.json\\") },
{ pkg: \\"dom-serializer\\", path: \\"/node_modules/dom-serializer/lib/index.d.ts\\", source: require(\\"!!raw-loader!/node_modules/dom-serializer/lib/index.d.ts\\").default },
];
"
`;
